lib/api/api.dart

```diff
@@ -779,4 +793,10 @@ class ApiSingleton {
       return false;
     }
   }
+  
+  Future<String> getCityNameFromCoords(double lat, double lon) async {
```

Comment:
I think it'd be better to place this function in the Utils folder since it's note related to how to communicate with the API.